test/test_utils/__init__.py

```diff
@@ -655,6 +655,20 @@ def get_framework_and_version_from_tag(image_uri):
     return tested_framework, tag_framework_version
 
 
+def get_cuda_version_from_tag(image_uri):
+    """
+    Return the cuda version from the image tag.
+    :param image_uri: ECR image URI
+    :return: cuda version
+    """
+    cuda_framework_version = None
+
+    cuda_str = ["cu", "gpu"]
+    if all(keyword in image_uri for keyword in cuda_str):
+        cuda_framework_version = image_uri.split(":")[-1].split("-")[3]
```

Comment:
Please use regular expressions to parse the cuda version out of the image uri, because the "string split and index" method is extremely flaky.